docs/modules/ROOT/pages/guides/erc20-supply.adoc

```diff
@@ -0,0 +1,139 @@
+= Creating ERC20 Supply
+
+:eip-20: https://eips.ethereum.org/EIPS/eip-20[EIP20]
+
+In this guide, you will learn how to create an ERC20 token with a custom supply mechanism.
+We will showcase the creation of both a fixed supply and a dynamic supply that you will be able to apply to your own smart contract development practice.
+
+The standard interface implemented by tokens built on Starknet comes from the popular token standard on Ethereum called ERC20.
+{eip-20}, from which the ERC20 contract is derived, does not specify how tokens are created.
+We'll go over two strategies for creating token supply.
+
+== Fixed Supply
+
+Let's say we want to create a token named MyToken with a fixed token supply.
+We can achieve this by setting the token supply in the constructor which will execute upon deployment.
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyToken {
+    use starknet::ContractAddress;
+    use openzeppelin::token::erc20::ERC20;
+
+    #[storage]
+    struct Storage {}
+
+    #[constructor]
+    fn constructor(
+        self: @ContractState,
+        initial_supply: u256,
+        recipient: ContractAddress
+    ) {
+        let name = 'MyToken';
+        let symbol = 'MTK';
+
+        let mut unsafe_state = ERC20::unsafe_new_contract_state();
+        ERC20::InternalImpl::initializer(ref unsafe_state, name, symbol);
+        ERC20::InternalImpl::_mint(ref unsafe_state, recipient, initial_supply);
+    }
+}
+----
+
+In the constructor, we're first grabbing the ERC20 state and calling the ERC20 initializer to set the token name and symbol.
+Next, we're calling the internal `_mint` function which creates `initial_supply` of tokens and allocates them to `recipient`.
+Since the internal `_mint` is not exposed in the ERC20 library, it will not be possible to create any more tokens.
+In other words, we've implemented a fixed token supply!
+
+== Dynamic Supply
+
+:access-control: xref:/access.adoc[Access Control]
+
+ERC20 contracts with a dynamic supply must include a mechanism for creating (and potentially destroying) tokens.
```

Comment:
```suggestion
ERC20 contracts with a dynamic supply include a mechanism for creating or destroying tokens.
```
it's a definition rather than a requirement 😅